### ISSUE TITLE
Tips for training with randomly initialised word embeddings

### ISSUE THREAD

**Issue by tannonk** [pytorch/fairseq#3015]
I'm working with Fairseq's transformer model for a seq2seq translation-style task where the source language is the same as the target language (e.g. de-de). However, when I initialise my word embeddings randomly (i.e. without specifying a path to a pre-trained embedding file), my model fails to train. 

After following the example https://github.com/pytorch/fairseq/tree/master/examples/translation#iwslt14-german-to-english-transformer, I applied the same preprocessing steps to my own data:

	1. tokenize
	2. learn and apply bpe
	3. binarize with `fairseq-preprocess`

Then, for training, I'm using the command:

```
fairseq-train $DATA_BIN \
--dataset-impl mmap \
--arch transformer \
--task translation \
--lr 5e-4 --optimizer adam \
--dropout 0.3 --attention-dropout 0.3 --activation-dropout 0.3 \
--max-epoch 10 --max-tokens 3360 --update-freq 2 
--share-all-embeddings \
--skip-invalid-size-inputs-valid-test \
--save-dir $CHECKPOINTS
```

This command differs slightly from that used in the IWSLT14 example, but I figured overall functionality should be the same.

According to the output training logs, for my data, train loss fails to improve, while validation loss jumps all over the place. Meanwhile, for the IWSLT14 example data (with the train command from the example), both train and validation loss and perplexity begin to quickly decrease.

If I initialise the embedding matrix for my data from pre-trained embeddings, say BPEmbs (https://nlp.h-its.org/bpemb/), training/validation loss and perplexity decrease nicely over the first epoch.

As far as I understand, the default behaviour is to learn the embedding matrix along with all other model weights, so I'm not sure why it's failing to learn them when training on my data. Am I missing something regarding training with randomly initialised embeddings vs. pre-trained embeddings?

Thanks in advance!

**Comment by ahmed-gamal97** [comment]
Hi @tannonk 

I wonder how you used pre-trained embeddings for training a machine translation model. I have a trained XLM-roberta encoder model and I would like to know if I can use it is weights as embedding for a machine translation task.

Thanks in Advance

**Comment by tannonk** [comment]
Hi @ahmed-gamal97 

I'm not sure how relevant this is to your situation, but I was providing static pre-trained embeddings using the `--encoder-embed-path`  and `--decoder-embed-path` arguments and providing the path to locally stored embeddings as text files, e.g. `de.wiki.bpe.vs10000.d200.w2v.txt` which can be downloaded from https://github.com/bheinzerling/bpemb
